File: jApp/src/main/java/j/s/yarlykov/ui/fragmentbased/SensorInfoDialog.java
package j.s.yarlykov.ui.fragmentbased;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import j.s.yarlykov.R;

public class SensorInfoDialog extends DialogFragment implements View.OnClickListener {
    private String content = null;
    private final static String key = "key";

    public static SensorInfoDialog create(String data) {
        SensorInfoDialog dialog = new SensorInfoDialog();

        Bundle args = new Bundle();
        args.putString(key, data);
        dialog.setArguments(args);
        return dialog;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        content = getArguments().getString(key);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        getDialog().setTitle(getString(R.string.sensorInfo));
        return inflater.inflate(R.layout.sensor_info_dialog, null);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        view.findViewById(R.id.btnCloseDialog).setOnClickListener(this);

        if(content != null) {
            ((TextView)view.findViewById(R.id.tvSensorInfo))
                    .setText(content);
        }
    }

    @Override
    public void onClick(View v) {
        dismiss();
    }
}

File: kApp/src/main/java/k/s/yarlykov/ui/fragmentbased/SensorInfoDialog.kt
package k.s.yarlykov.ui.fragmentbased

import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import k.s.yarlykov.R
import kotlinx.android.synthetic.main.sensor_info_dialog.*

class SensorInfoDialog: DialogFragment() {

    var content: String? = null

    companion object {
        const val key: String = "key"

        fun create(data: String): SensorInfoDialog {

            return SensorInfoDialog().apply {
                arguments = Bundle().apply {
                    putString(key, data)
                }
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        content = arguments?.getString(key)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        dialog.setTitle(getString(R.string.sensorInfo))
        return inflater.inflate(R.layout.sensor_info_dialog, null)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        btnCloseDialog.setOnClickListener { dismiss() }

        content?.let {
            tvSensorInfo.text = it
        }
    }

}